app/assets/javascripts/comments.js

```diff
@@ -129,6 +79,30 @@ $(() => {
     }
   });
 
+  $(document).on('click', '.js--restrict-thread, .js--unrestrict-thread', async evt => {
+    evt.preventDefault();
+
+    const $tgt = $(evt.target);
+    const threadID = $tgt.data("thread")
+    const action = $tgt.data("action")
+    const route = $tgt.hasClass("js--restrict-thread") ? 'restrict' : 'unrestrict';
+
+    const resp = await fetch(`/comments/thread/${threadID}/${route}`, {
+      method: 'POST',
+      credentials: 'include',
+      headers: { 'X-CSRF-Token': QPixel.csrfToken(), 'Content-Type': 'application/x-www-form-urlencoded; charset=utf-8' },
```

Comment:
**suggestion** (code-clarity): Usually more clear to use `application/json` - Rails will accept it either way, but means you can use `JSON.stringify({ your: 'data', goes: 'here' })` for the `body`.